src/goto-symex/goto_symex_state.cpp

```diff
@@ -622,16 +622,20 @@ static bool requires_renaming(const typet &type, const namespacet &ns)
     return requires_renaming(array_type.subtype(), ns) ||
            !array_type.size().is_constant();
   }
-  else if(
-    type.id() == ID_struct || type.id() == ID_union || type.id() == ID_class)
+  else if(type.id() == ID_struct || type.id() == ID_union)
```

Comment:
Is this change related to the rest of the PR?

Reply:
I guess I sneaked that one in - it just cleans up a dead branch that I had complained about back in https://github.com/diffblue/cbmc/pull/3633#discussion_r244408981. The agreed fix just never made it into this part of the code...

Reply:
It would be better in a separate commit

Reply:
Fair enough - that's now done.